test/e2e/lib/pages/frontend/comments-area-component.js

```diff
@@ -16,50 +15,46 @@ export default class CommentsAreaComponent extends AsyncBaseContainer {
 	}
 
-	async _postComment( { comment } ) {
-		const commentForm = By.css( '#commentform' );
+	/**
+	 * Posts the first comment to a post (under "Join the Conversation" in the footer).
+	 *
+	 * @param { string } comment The comment text.
+	 */
+	async _postComment( comment ) {
 		const commentField = By.css( '#comment' );
 		const submitButton = By.css( '.form-submit #comment-submit' );
-		const commentContent = By.xpath( `//div[@class='comment-content']/p[.='${ comment }']` );
 
-		await driverHelper.scrollIntoView( this.driver, commentForm, 'end' );
-		await driverHelper.clickWhenClickable( this.driver, commentForm );
-		await driverHelper.scrollIntoView( this.driver, commentForm, 'end' );
-		await this.switchToFrameIfJetpack();
-
-		await driverHelper.clickWhenClickable( this.driver, commentForm );
-		await driverHelper.waitUntilLocatedAndVisible( this.driver, submitButton );
 		await driverHelper.scrollIntoView( this.driver, submitButton );
 		await driverHelper.setWhenSettable( this.driver, commentField, comment );
 		await driverHelper.clickWhenClickable( this.driver, submitButton );
-		await this.driver.switchTo().defaultContent();
-		return await driverHelper.waitUntilLocatedAndVisible( this.driver, commentContent );
 	}
 
-	async reply( commentObj, depth = 2 ) {
+	/**
+	 * Posts a reply to a comment. Re-uses most of @see _postComment, but starts
+	 * the action by clicking the "Reply" link for an existing comment.
+	 *
+	 * @param {*} comment The comment text.
+	 */
+	async reply( comment ) {
 		const replyButton = By.css( '.comment-reply-link' );
-		const replyContent = By.xpath(
-			`//li[contains(@class,'depth-${ depth }')]//div[@class='comment-content']/p[.='${ commentObj.comment }']`
-		);
+
 		await driverHelper.clickWhenClickable( this.driver, replyButton );
-		await this._postComment( commentObj );
-		return await driverHelper.waitUntilLocatedAndVisible( this.driver, replyContent );
+		await this._postComment( comment );
 	}
 
-	async switchToFrameIfJetpack() {
-		if ( dataHelper.getJetpackHost() === 'WPCOM' ) {
-			return false;
-		}
-		await this.driver.sleep( 1000 );
-
-		const iFrameSelector = By.css( 'iframe.jetpack_remote_comment' );
-
-		await this.driver.switchTo().defaultContent();
-		await driverHelper.waitUntilLocatedAndVisible( this.driver, iFrameSelector );
-		await this.driver.wait(
-			until.ableToSwitchToFrame( iFrameSelector ),
-			this.explicitWaitMS,
-			'Could not switch to comment form iFrame'
+	/**
+	 * Verifies if a comment is visible. If you tweak the depth, you can also
+	 * "point" it to a reply comment.
+	 *
+	 * @param { string } comment The comment text.
+	 * @param { number } depth How nested you want to check. The first comment has
+	 * a depth of 1, while the first reply to it would have a depth of 2.
+	 */
+	async verifyCommentIsVisible( comment, depth = 1 ) {
+		const commentLocator = By.xpath(
+			`//li[contains(@class,'depth-${ depth }')]//div[@class='comment-content']/p[.='${ comment }']`
```

Comment:
There wasn't a need for two slightly similar xpath locators in two different functions, so I extracted the logic into this single helper. Also, I thought it'd best not to couple the responsibility of asserting the result with the side-effect functions (`_postComment` and `reply`). Each test now has the responsibility of asserting the result by calling this function, see [this line](https://github.com/Automattic/wp-calypso/pull/52217/files#diff-0b7e3b134038ca1288f2be96f6db46367a456909d110cbc9c2b9c7b80785b522R60) as an example.